pkg/image/api/validation/validation.go

```diff
@@ -43,6 +44,17 @@ func ValidateImageStream(stream *api.ImageStream) fielderrors.ValidationErrorLis
 
 	result = append(result, validation.ValidateObjectMeta(&stream.ObjectMeta, true, MinimalNameValidation).Prefix("metadata")...)
 
+	// Ensure we can generate a valid docker image repository from namespace/name
```

Comment:
Since they can switch between internal/external post-creation, I'd rather keep this "strict" for now